File: dotnet/src/EPostak/Resources/ConnectorAdvancedResource.cs
using EPostak.Models;

namespace EPostak.Resources;

/// <summary>
/// Explicit opt-in surface for lower-level and compatibility Connector APIs.
/// Most ERP integrations only need
/// <c>Connector.Customers.For(customerRef).Documents</c> and <c>Events</c>.
/// </summary>
public sealed class ConnectorAdvancedResource
{
    private readonly ConnectorResource _connector;

    internal ConnectorAdvancedResource(ConnectorResource connector)
    {
        _connector = connector;
        Documents = new ConnectorAdvancedDocumentsResource(connector);
    }

    /// <summary>UBL and evidence artifacts kept outside the golden business flow.</summary>
    public ConnectorAdvancedDocumentsResource Documents { get; }


    public Task<ConnectorPreflightResponse> PreflightAsync(ConnectorPreflightRequest request, CancellationToken ct = default)
        => _connector.PreflightAsync(request, ct);

    public Task<ConnectorSendResponse> SendAsync(ConnectorSendRequest request, CancellationToken ct = default)
        => _connector.SendAsync(request, ct);

    public Task<ConnectorSendResponse> SendAsync(ConnectorSendRequest request, string? idempotencyKey, CancellationToken ct = default)
        => _connector.SendAsync(request, idempotencyKey, ct);

    public Task<ConnectorStatusResponse> StatusAsync(string documentId, CancellationToken ct = default)
        => _connector.StatusAsync(documentId, ct);

    public Task<ConnectorInboxListResponse> InboxAsync(ConnectorListParams? @params = null, CancellationToken ct = default)
        => _connector.InboxAsync(@params, ct);

    public Task<ConnectorInboxDocument> GetInboxDocumentAsync(string documentId, CancellationToken ct = default)
        => _connector.GetInboxDocumentAsync(documentId, ct);

    public Task<ConnectorAckResponse> AckAsync(string documentId, CancellationToken ct = default)
        => _connector.AckAsync(documentId, ct);

    /// <summary>
    /// Compatibility event polling. Prefer
    /// <c>Connector.Customers.For(customerRef).Events.ListAsync(...)</c> for the
    /// customer-scoped business feed.
    /// </summary>
    public Task<ConnectorEventsResponse> EventsAsync(ConnectorListParams? @params = null, CancellationToken ct = default)
        => _connector.EventsAsync(@params, ct);

    public Task<ConnectorOutboxStageResponse> StageOutboxAsync(ConnectorOutboxStageRequest request, CancellationToken ct = default)
        => _connector.StageOutboxAsync(request, ct);

    public Task<ConnectorOutboxListResponse> ListOutboxAsync(ConnectorOutboxListParams? @params = null, CancellationToken ct = default)
        => _connector.ListOutboxAsync(@params, ct);

    public Task<ConnectorOutboxItem> GetOutboxItemAsync(string outboxId, CancellationToken ct = default)
        => _connector.GetOutboxItemAsync(outboxId, ct);

    public Task<ConnectorOutboxItem> SendOutboxItemAsync(string outboxId, ConnectorOutboxSendOptions? options = null, CancellationToken ct = default)
        => _connector.SendOutboxItemAsync(outboxId, options, ct);

    public Task<ConnectorOutboxBatchSendResponse> SendOutboxBatchAsync(ConnectorOutboxBatchSendRequest? request = null, CancellationToken ct = default)
        => _connector.SendOutboxBatchAsync(request, ct);

    public Task<ConnectorOutboxItem> CancelOutboxItemAsync(string outboxId, CancellationToken ct = default)
        => _connector.CancelOutboxItemAsync(outboxId, ct);

    public Task<ConnectorAutopilotRunResponse> AutopilotAsync(ConnectorAutopilotRequest request, CancellationToken ct = default)
        => _connector.AutopilotAsync(request, ct);

    public Task<Dictionary<string, object?>> MapperAsync(ConnectorMapperRequest request, CancellationToken ct = default)
        => _connector.MapperAsync(request, ct);

    public Task<ConnectorAutopilotRunResponse> ZenInputAsync(ConnectorZenInputRequest request, CancellationToken ct = default)
        => _connector.ZenInputAsync(request, ct);

    public Task<ConnectorAutopilotRunResponse> GetAutopilotRunAsync(string autopilotId, CancellationToken ct = default)
        => _connector.GetAutopilotRunAsync(autopilotId, ct);

    public Task<ConnectorAutopilotRunResponse> SendAutopilotRunAsync(string autopilotId, CancellationToken ct = default)
        => _connector.SendAutopilotRunAsync(autopilotId, ct);

    public Task<ConnectorReconcileResponse> ReconcileAsync(ConnectorReconcileParams? @params = null, CancellationToken ct = default)
        => _connector.ReconcileAsync(@params, ct);

    public Task<ConnectorMailboxListResponse> MailboxesAsync(CancellationToken ct = default)
        => _connector.MailboxesAsync(ct);

    public Task<Dictionary<string, object?>> RepairMailboxAsync(ConnectorMailboxRepairRequest? request = null, CancellationToken ct = default)
        => _connector.RepairMailboxAsync(request, ct);

    public Task<ConnectorMailboxUpdateResponse> UpdateMailboxSendPolicyAsync(string customerRef, ConnectorSendPolicyOptions request, CancellationToken ct = default)
        => _connector.UpdateMailboxSendPolicyAsync(customerRef, request, ct);

    public Task<ConnectorSyncResponse> SyncAsync(ConnectorSyncParams? @params = null, CancellationToken ct = default)
        => _connector.SyncAsync(@params, ct);

    public Task<ConnectorActionResponse> RunActionAsync(string actionId, ConnectorActionRequest? request = null, CancellationToken ct = default)
        => _connector.RunActionAsync(actionId, request, ct);

}

File: java/src/main/java/sk/epostak/sdk/resources/ConnectorAdvancedResource.java
package sk.epostak.sdk.resources;

import sk.epostak.sdk.models.*;

import java.util.Map;

/**
 * Explicit opt-in surface for lower-level and compatibility Connector APIs.
 * <p>
 * Most integrations only need
 * {@code connector().customers().forCustomer(customerRef).documents()} and
 * {@code events()}. Use this resource for legacy firm-scoped delivery or for
 * advanced orchestration such as outbox, Autopilot, mapper, reconcile,
 * mailbox, sync, and action execution.
 */
@SuppressWarnings("deprecation")
public final class ConnectorAdvancedResource {
    private final ConnectorResource connector;
    private final ConnectorAdvancedDocumentsResource documents;

    ConnectorAdvancedResource(ConnectorResource connector) {
        this.connector = connector;
        this.documents = new ConnectorAdvancedDocumentsResource(connector);
    }

    public ConnectorAdvancedDocumentsResource documents() { return documents; }

    public ConnectorPreflightResponse preflight(ConnectorPreflightRequest request) {
        return connector.preflight(request);
    }

    public ConnectorSendResponse send(Map<String, Object> request) {
        return connector.send(request);
    }

    public ConnectorSendResponse send(Map<String, Object> request, String idempotencyKey) {
        return connector.send(request, idempotencyKey);
    }

    public ConnectorStatusResponse status(String documentId) {
        return connector.status(documentId);
    }

    public ConnectorInboxListResponse inbox(ConnectorListParams params) {
        return connector.inbox(params);
    }

    public ConnectorInboxListResponse inbox() {
        return connector.inbox();
    }

    public ConnectorInboxDocument getInboxDocument(String documentId) {
        return connector.getInboxDocument(documentId);
    }

    public ConnectorAckResponse ack(String documentId) {
        return connector.ack(documentId);
    }

    /** Firm-scoped technical event feed with legacy {@code status} events. */
    public ConnectorEventsResponse events(ConnectorListParams params) {
        return connector.events(params);
    }

    public ConnectorEventsResponse events() {
        return connector.events();
    }

    public ConnectorOutboxStageResponse stageOutbox(ConnectorOutboxStageRequest request) {
        return connector.stageOutbox(request);
    }

    public ConnectorOutboxListResponse listOutbox(ConnectorOutboxListParams params) {
        return connector.listOutbox(params);
    }

    public ConnectorOutboxListResponse listOutbox() {
        return connector.listOutbox();
    }

    public ConnectorOutboxItem getOutboxItem(String outboxId) {
        return connector.getOutboxItem(outboxId);
    }

    public ConnectorOutboxItem sendOutboxItem(String outboxId, ConnectorOutboxSendOptions options) {
        return connector.sendOutboxItem(outboxId, options);
    }

    public ConnectorOutboxItem sendOutboxItem(String outboxId) {
        return connector.sendOutboxItem(outboxId);
    }

    public ConnectorOutboxBatchSendResponse sendOutboxBatch(ConnectorOutboxBatchSendRequest request) {
        return connector.sendOutboxBatch(request);
    }

    public ConnectorOutboxBatchSendResponse sendOutboxBatch() {
        return connector.sendOutboxBatch();
    }

    public ConnectorOutboxItem cancelOutboxItem(String outboxId) {
        return connector.cancelOutboxItem(outboxId);
    }

    public ConnectorAutopilotRunResponse autopilot(ConnectorAutopilotRequest request) {
        return connector.autopilot(request);
    }

    public Map<String, Object> mapper(Map<String, Object> request) {
        return connector.mapper(request);
    }

    public ConnectorAutopilotRunResponse zenInput(Map<String, Object> request) {
        return connector.zenInput(request);
    }

    public ConnectorAutopilotRunResponse getAutopilotRun(String autopilotId) {
        return connector.getAutopilotRun(autopilotId);
    }

    public ConnectorAutopilotRunResponse sendAutopilotRun(String autopilotId) {
        return connector.sendAutopilotRun(autopilotId);
    }

    public ConnectorReconcileResponse reconcile(ConnectorReconcileParams params) {
        return connector.reconcile(params);
    }

    public ConnectorReconcileResponse reconcile() {
        return connector.reconcile();
    }

    public ConnectorMailboxListResponse mailboxes() {
        return connector.mailboxes();
    }

    public Map<String, Object> repairMailbox(ConnectorMailboxRepairRequest request) {
        return connector.repairMailbox(request);
    }

    public Map<String, Object> repairMailbox() {
        return connector.repairMailbox();
    }

    public ConnectorMailboxUpdateResponse updateMailboxSendPolicy(
            String customerRef,
            ConnectorSendPolicyOptions request
    ) {
        return connector.updateMailboxSendPolicy(customerRef, request);
    }

    public ConnectorSyncResponse sync(ConnectorSyncParams params) {
        return connector.sync(params);
    }

    public ConnectorSyncResponse sync() {
        return connector.sync();
    }

    public ConnectorActionResponse runAction(String actionId, ConnectorActionRequest request) {
        return connector.runAction(actionId, request);
    }

    public ConnectorActionResponse runAction(String actionId) {
        return connector.runAction(actionId);
    }
}
